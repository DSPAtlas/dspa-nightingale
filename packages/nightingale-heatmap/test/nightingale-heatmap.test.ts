import NightingaleHeatmap from "../dist/index";
import * as data from "./contact-map.json";
let rendered: NightingaleHeatmap;

describe("nightingale-navigation tests", () => {
  beforeEach(() => {
    rendered = new NightingaleHeatmap();
    rendered.setAttribute("height", "200");
    rendered.setAttribute("height", "200");
    document.documentElement.appendChild(rendered);
    // eslint-disable-next-line @typescript-eslint/no-explicit-any
    (rendered as any).data = data.value;
  });

  afterEach(() => {
    document.documentElement.removeChild(rendered);
  });

  test("it should display the heatmap correctly", async () => {
    expect(rendered).toMatchSnapshot();
    expect(rendered.data.slice(-5)).toMatchSnapshot();
  });

  test("data for the heatmap correctly after symmetric", () => {
    rendered.setAttribute("symmetric", "symmetric");
    return new Promise((done) => {
      window.requestAnimationFrame(() => {
        expect(rendered.data.slice(-5)).toMatchSnapshot();
        done(true);
      });
    });
  });
});
