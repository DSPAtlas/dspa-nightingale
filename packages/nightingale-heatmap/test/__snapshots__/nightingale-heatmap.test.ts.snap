// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nightingale-navigation tests data for the heatmap correctly after symmetric 1`] = `
[
  [
    151,
    153,
    0.1983642578125,
  ],
  [
    152,
    152,
    0.01183319091796875,
  ],
  [
    152,
    152,
    0,
  ],
  [
    152,
    153,
    0.94921875,
  ],
  [
    153,
    153,
    0.03485107421875,
  ],
]
`;

exports[`nightingale-navigation tests it should display the heatmap correctly 1`] = `
<nightingale-heatmap
  height="200"
  style="width: 100%; display: block; line-height: 0;"
>
  <!---->
  
      
  <canvas
    class="canvas-heatmap"
    height="200"
    style="position: absolute; margin-left: 10px; margin-top: 0px;"
    width="80"
  />
  <svg
    class="svg-heatmap"
    height="200"
    style="position: absolute; pointer-events: none;"
    width="100"
  >
    
        
    <g
      class="svg-heatmap-group"
      transform="translate(10,0)"
    >
      
          
      <g
        class="x-axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(0,200)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <line
            stroke="currentColor"
            y2="3"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="6"
          >
            1
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(15.76315789473684,0)"
        >
          <line
            stroke="currentColor"
            y2="3"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="6"
          >
            30
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(31.552631578947366,0)"
        >
          <line
            stroke="currentColor"
            y2="3"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="6"
          >
            60
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(47.34210526315789,0)"
        >
          <line
            stroke="currentColor"
            y2="3"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="6"
          >
            90
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(63.13157894736842,0)"
        >
          <line
            stroke="currentColor"
            y2="3"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="6"
          >
            120
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(78.92105263157895,0)"
        >
          <line
            stroke="currentColor"
            y2="3"
          />
          <text
            dy="0.71em"
            fill="currentColor"
            y="6"
          >
            150
          </text>
        </g>
      </g>
      
          
      <text
        class="x-label"
        style="text-anchor:middle"
        transform="translate(40,200)"
      >
        Residue
      </text>
      
          
      <g
        class="y-axis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(0,0.6578947368421053)"
        >
          <line
            stroke="currentColor"
            x2="-3"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            1
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,38.81578947368421)"
        >
          <line
            stroke="currentColor"
            x2="-3"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            30
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,78.28947368421053)"
        >
          <line
            stroke="currentColor"
            x2="-3"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            60
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,117.76315789473685)"
        >
          <line
            stroke="currentColor"
            x2="-3"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            90
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,157.23684210526318)"
        >
          <line
            stroke="currentColor"
            x2="-3"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            120
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,196.71052631578948)"
        >
          <line
            stroke="currentColor"
            x2="-3"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-6"
          >
            150
          </text>
        </g>
      </g>
      
          
      <g
        class="hovered-area"
      />
      
          
      <text
        class="y-label"
        dy="1em"
        style="text-anchor:middle"
        transform="rotate(-90)"
        x="-100"
        y="-10"
      >
        Residue
      </text>
      
        
    </g>
    
      
  </svg>
  
    
</nightingale-heatmap>
`;

exports[`nightingale-navigation tests it should display the heatmap correctly 2`] = `
[
  [
    153,
    149,
    0,
  ],
  [
    153,
    150,
    0,
  ],
  [
    153,
    151,
    0,
  ],
  [
    153,
    152,
    0,
  ],
  [
    153,
    153,
    0.03485107421875,
  ],
]
`;
